In [23]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt
import json
from sklearn.preprocessing import StandardScaler

In [24]:
df = pd.read_csv('cleanDF.csv', index_col = 0, parse_dates = ['publish_time'])

In [25]:
df.head()

video_id trending_date              publish_time    views   likes  \
0  2kyS6SvSYSE    2017-11-14 2017-11-13 17:13:01+00:00   748374   57527   
1  1ZAPwfrtAFY    2017-11-14 2017-11-13 07:30:00+00:00  2418783   97185   
2  5qpjK5DgCt4    2017-11-14 2017-11-12 19:05:24+00:00  3191434  146033   
3  puqaWrEC7tY    2017-11-14 2017-11-13 11:00:04+00:00   343168   10172   
4  d380meD0W0M    2017-11-14 2017-11-12 18:01:41+00:00  2095731  132235   

   dislikes  comment_count  category_id        Category  
0      2966          15954           22  People & Blogs  
1      6146          12703           24   Entertainment  
2      5339           8181           23          Comedy  
3       666           2146           24   Entertainment  
4      1989          17518           24   Entertainment

In [26]:
names = ['views', 'likes','dislikes', 'comment_count']

In [27]:
df1 = df[['views', 'likes','dislikes', 'comment_count']]

In [28]:
scaler = StandardScaler()

In [29]:
scaled = scaler.fit_transform(df1)

In [30]:
scaledDf = pd.DataFrame(scaled, columns = names)

In [31]:
scaledDf.head()

views     likes  dislikes  comment_count
0 -0.218069 -0.073137 -0.025677       0.200566
1  0.007844  0.100131  0.083867       0.113711
2  0.112341  0.313551  0.056067      -0.007101
3 -0.272871 -0.280033 -0.104908      -0.168336
4 -0.035847  0.253267 -0.059333       0.242351

In [32]:
X = df1[['views', 'dislikes', 'comment_count']]

In [33]:
y = df1[['likes']]

In [34]:
from sklearn.model_selection import train_test_split

In [35]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

MODELING

In [36]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score,mean_squared_error
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import MinMaxScaler
from sklearn.model_selection  import GridSearchCV
from sklearn.metrics import accuracy_score

In [37]:
transfer = MinMaxScaler((-1,1))
X_train = transfer.fit_transform(X_train)
X_test = transfer.transform(X_test)

In [38]:
lgr=LinearRegression(fit_intercept=True)
fit_model=lgr.fit(X_train,y_train)
prediction=lgr.predict(X_test)

print(fit_model.score(X_test,y_test))
print(r2_score(y_test,prediction))

0.8922147343196112
0.8922147343196112


In [39]:
from sklearn.ensemble import RandomForestClassifier

In [40]:
model1 = RandomForestClassifier()

In [41]:
model1.fit(X_train[:2000],np.ravel(y_train[:2000]))

RandomForestClassifier()

In [42]:
Y_pred_rf=model1.predict(X_test)

In [43]:
RandomForestClassifierScore = model1.score(X_test,y_test)
print("Accuracy obtained for Random Forest Classifier :",RandomForestClassifierScore*100)

Accuracy obtained for Method 2 - Random Forest Classifier : 0.463980463980464


In [44]:
ac = accuracy_score(y_test, Y_pred_rf)
ac

0.00463980463980464

In [45]:
from sklearn.tree import DecisionTreeClassifier
tree = DecisionTreeClassifier()
tree.fit(X_train[:2000],np.ravel(y_train[:2000]))
y_pred = tree.predict(X_test)

In [46]:
DecisionTreeClassifierScore = tree.score(X_test,y_test)
print("Accuracy obtained by Decision Tree Classifier model:",DecisionTreeClassifierScore*100)

Accuracy obtained by Decision Tree Classifier model: 0.47212047212047215


In [55]:
from sklearn import tree, metrics
print("Accuracy:", metrics.accuracy_score(y_test,y_pred))

Accuracy: 0.0047212047212047215


In [48]:
from sklearn.neighbors import KNeighborsClassifier
KNN_model = KNeighborsClassifier(n_neighbors=10, metric='manhattan', weights='distance')
KNN_model.fit(X_train, np.ravel(y_train))

KNeighborsClassifier(metric='manhattan', n_neighbors=10, weights='distance')

In [49]:
KNN_model.score(X_test, y_test)

0.015466015466015465

In [51]:
from sklearn.svm import SVC

svm = SVC(kernel='linear')
svm.fit(X_train[:2000],np.ravel(y_train[:2000]))

# Predict using model:

y_predict_svm=svm.predict(X_test)

Accuracy_svm=svm.score(X_test,y_test)
print(Accuracy_svm)

0.004151404151404151


The best model with the greatest accuracy would be the lienar regression model at 89% accurate. The other models struggled to attain accuracy due to the nature of how those models work in a regression analysis. 